kolibri/core/decorators.py

```diff
@@ -275,3 +276,19 @@ def wrapped_view(*args, **kwargs):
         return view_func(*args, **kwargs)
     wrapped_view.signin_redirect_exempt = True
     return wraps(view_func)(wrapped_view)
+
+
+def cache_no_user_data(view_func):
+    """
+    Set appropriate Vary on headers on a view that specify there is
+    no user specific data being rendered in the view.
+    """
+    def inner_func(*args, **kwargs):
+        request = args[0]
+        del request.session
```

Comment:
Do we want a decorator that just deletes the entire session data? Where will this be used? Normally, it would seem like a logout procedure. This doesn't match that well with the `__docstring__` of the decorator.

Reply:
I understand the utility of this decorator except for the point Ben comments, why is it doing `del request.session`?
All the other lines in the decorator make sense for me, but I don't understand the rationale of deleting the session when switching between learn/learn.html and user/user.html

Reply:
The reason for deleting the session is that otherwise Django adds a Vary header for cookies onto the request, meaning no caching takes place.

I could achieve the same effect by having this decorator add a "don't add cookie vary header" flag to the view, and then modify other session modifying middleware to obey that directive, but this seemed simplest.